### PR TITLE
[MRG] auto replace gbdt::prediction with if-else prediction

### PR DESCRIPTION
**Pull request by wxchan**

### CODE REVIEW

**Comment by guolinke** [comment]
@wxchan I suggest not just replace the ```gbdt_prediction.cpp```. 
you can use a macro ```use_hard_code_prediction``` to control use which one.

**Comment by wxchan** [comment]
@guolinke ok, I will try.

File: .travis.yml [hunk 1]
```diff
@@ -53,6 +53,11 @@ script:
 - cd $TRAVIS_BUILD_DIR/tests/c_api_test && python test.py
 - cd $TRAVIS_BUILD_DIR/python-package && python setup.py install
 - cd $TRAVIS_BUILD_DIR/tests/python_package_test && python test_basic.py && python test_engine.py && python test_sklearn.py && python test_plotting.py
+- cd $TRAVIS_BUILD_DIR
+- rm -rf build && mkdir build && cd build && cmake .. && make
+- cd $TRAVIS_BUILD_DIR/tests/cpp_test && ../../lightgbm config=train.conf && ../../lightgbm config=predict.conf output_result=origin.pred
+- cd $TRAVIS_BUILD_DIR/build && make clean && make
+- cd $TRAVIS_BUILD_DIR/tests/cpp_test && ../../lightgbm config=predict.conf output_result=ifelse.pred && python test.py
```
Comment (wxchan) [inline .travis.yml]: @guolinke I am not so sure how to do it, do you mean use some option like USE_HARD_CODE in CMakeLists.txt? and what should the default generated codes name and locate? Can you give a logic how to test it like these lines?

Comment (guolinke) [inline .travis.yml]: generate them in  the `gbdt_prediction.cpp`
code in `gbdt_prediction.cpp` may like:

```
// p1 
#ifndef USE_HARD_CODE 
original prediction code
#else
gerenrated hard code
#endif

```

For the macro `USE_HARD_CODE`, maybe you can define it in the `p1` place when generate code. So user can comment it to use the original prediction code.

Comment (wxchan) [inline .travis.yml]: @guolinke got it. My current logic is, if convert_model file exists, put it into `original prediction code`, else only generate hard code.

**Pull request merged**